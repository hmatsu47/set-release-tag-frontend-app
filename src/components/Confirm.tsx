import { Show } from "solid-js";
import { useTheme } from "@suid/material";
import Backdrop from "@suid/material/Backdrop";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Stack from "@suid/material/Stack";
import Typography from "@suid/material/Typography";
import { updateImages } from "../api/updateImages";
import { isOpenedConfirm, setIsOpenedConfirm } from "../signal";

export const Confirm = () => {
  const theme = useTheme();

  return (
    <Show when={isOpenedConfirm() === true} fallback={<></>}>
      <Backdrop
        sx={{ color: "#fff", zIndex: theme.zIndex.drawer + 1 }}
        open={isOpenedConfirm()}
      ></Backdrop>
      <Box
        sx={{
          position: "absolute",
          top: "25%",
          left: "50%",
          transform: "translate(-50%, -50%)",
          width: 400,
          bgcolor: theme.palette.background.paper,
          border: "2px solid #000",
          boxShadow: "24px",
          p: 4,
          zIndex: theme.zIndex.drawer + 1,
        }}
      >
        <Typography variant="subtitle1">
          本番リリース時に使用するイメージを選択します。
        </Typography>
        <Typography variant="subtitle1">よろしいですか？</Typography>
        <Box sx={{ paddingTop: "20px" }}></Box>
        <Stack spacing={4} direction="row" sx={{ justifyContent: "center" }}>
          <Button
            variant="contained"
            size="small"
            color="primary"
            onClick={async (e) => {
              setIsOpenedConfirm(false);
              await updateImages();
            }}
            title="はい"
          >
            はい
          </Button>
          <Button
            variant="outlined"
            size="small"
            color="inherit"
            onClick={() => {
              setIsOpenedConfirm(false);
            }}
            title="いいえ"
          >
            いいえ
          </Button>
        </Stack>
      </Box>
    </Show>
  );
};
