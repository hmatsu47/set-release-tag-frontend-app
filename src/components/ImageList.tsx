import { For, Show, onMount } from "solid-js";
import Box from "@suid/material/Box";
import Button from "@suid/material/Button";
import Paper from "@suid/material/Paper";
import Table from "@suid/material/Table";
import TableBody from "@suid/material/TableBody";
import TableCell from "@suid/material/TableCell";
import TableContainer from "@suid/material/TableContainer";
import TableHead from "@suid/material/TableHead";
import TableRow from "@suid/material/TableRow";
import Typography from "@suid/material/Typography";
import { fetchImages } from "../api/fetchImages";
import { images, setIsOpenedConfirm, setSelectedTag } from "../signal";
import { formatDateTimeDisplay } from "../formatDateTime";

export const ImageList = () => {
  onMount(async () => {
    if (images() == undefined) {
      await fetchImages();
    }
  });

  return (
    <Show when={images() !== undefined} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Show
          when={images() !== null}
          fallback={
            <Typography variant="h6">
              リポジトリにイメージがありません
            </Typography>
          }
        >
          <TableContainer component={Paper}>
            <Table aria-label="simple table">
              <TableHead>
                <TableRow>
                  <TableCell>選択</TableCell>
                  <TableCell>イメージタグ</TableCell>
                  <TableCell>プッシュ日時</TableCell>
                  <TableCell>
                    サイズ
                    <br />
                    （MB）
                  </TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                <For each={images()} fallback={<></>}>
                  {(imageItem) => (
                    <TableRow>
                      <TableCell component="th" scope="row">
                        <Button
                          variant={
                            imageItem.tags.filter((n) => n === "release")
                              .length > 0
                              ? "contained"
                              : "outlined"
                          }
                          size="small"
                          color={
                            imageItem.tags.filter((n) => n === "release")
                              .length > 0
                              ? "primary"
                              : "inherit"
                          }
                          onClick={async (e) => {
                            if (
                              imageItem.tags.filter((n) => n === "release")
                                .length == 0
                            ) {
                              setSelectedTag(imageItem.tags[0]);
                              setIsOpenedConfirm(true);
                            }
                          }}
                          sx={{ textTransform: "none" }}
                          title={`${imageItem.tags.join(", ")} を選択`}
                        >
                          選択
                        </Button>
                      </TableCell>
                      <TableCell>{imageItem.tags.join(", ")}</TableCell>
                      <TableCell>
                        {formatDateTimeDisplay(new Date(imageItem.pushed_at))}
                      </TableCell>
                      <TableCell align="right">
                        {(
                          Math.round(
                            (imageItem.size / (1000 * 1000)) * Math.pow(10, 2)
                          ) / Math.pow(10, 2)
                        ).toLocaleString()}
                      </TableCell>
                    </TableRow>
                  )}
                </For>
              </TableBody>
            </Table>
          </TableContainer>
        </Show>
      </Box>
    </Show>
  );
};
