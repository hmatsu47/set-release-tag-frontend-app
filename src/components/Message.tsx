import { Show } from "solid-js";
import Alert from "@suid/material/Alert";
import Box from "@suid/material/Box";
import { message, messageSeverity } from "../signal";

export const Message = () => {
  return (
    <Show when={message()} fallback={<></>}>
      <Box
        sx={{
          width: "100%",
          minWidth: "1024px",
          display: "flex",
        }}
        aria-live="polite"
      >
        <Alert severity={messageSeverity()}>{message()}</Alert>
      </Box>
    </Show>
  );
};
