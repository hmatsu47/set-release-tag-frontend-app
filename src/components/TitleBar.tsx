import AppBar from "@suid/material/AppBar";
import Typography from "@suid/material/Typography";

export const TitleBar = () => {
  return (
    <AppBar position="static">
      <Typography
        variant="h6"
        component="div"
        sx={{
          flexGrow: 1,
          padding: "0 0 0 10px",
        }}
      >
        {localStorage.getItem("environment")
          ? `${localStorage.getItem("environment")}・`
          : ""}
        コンテナリリース設定
      </Typography>
    </AppBar>
  );
};
