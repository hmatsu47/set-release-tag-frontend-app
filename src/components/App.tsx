import { Component } from "solid-js";
import Box from "@suid/material/Box";
import Stack from "@suid/material/Stack";
import { ThemeProvider } from "@suid/material";
import { ColorTheme } from "../ColorTheme";
import { ImageList } from "./ImageList";
import { Message } from "./Message";
import { TitleBar } from "./TitleBar";
import { Confirm } from "./Confirm";

export const App: Component = () => {
  return (
    <ThemeProvider theme={ColorTheme}>
      <TitleBar />
      <Box
        sx={{
          padding: "10px 10px 0 10px",
          width: "100%",
          minWidth: "1024px",
          display: "flex",
          // justifyContent: "center",
        }}
        aria-live="polite"
      >
        <Stack spacing={2} direction="column">
          <Message />
          <ImageList />
          <Confirm />
        </Stack>
      </Box>
    </ThemeProvider>
  );
};
