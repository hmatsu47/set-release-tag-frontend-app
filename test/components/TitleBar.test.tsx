import { describe, expect, test } from "vitest";
import { render } from "@solidjs/testing-library";
import { formatSnapshot } from "../common/formatSnapshot";
import { TitleBar } from "../../src/components/TitleBar";

describe("<TitleBar />", () => {
  const titleBarList = [
    {
      title: "本番環境Postfix",
      environment: "本番環境Postfix",
      expected: "本番環境Postfix・コンテナリリース設定",
    },
    {
      title: "テスト環境Postfix",
      environment: "テスト環境Postfix",
      expected: "テスト環境Postfix・コンテナリリース設定",
    },
  ];
  titleBarList.forEach((testCase) => {
    beforeEach(() => {
      // localStorage はモック化すべきかも（うまくいかなかったため一旦モックなしで実装）
      localStorage.removeItem("environment");
    });
    test(testCase.title, async () => {
      if (testCase.environment) {
        localStorage.setItem("environment", testCase.environment);
      }
      const { container, findByText, unmount } = render(() => <TitleBar />);
      const expected = (await findByText(testCase.expected)) as HTMLElement;
      expect(expected).toHaveTextContent(testCase.expected);
      // css の名前が動的に変わるので固定値に置換
      const html = formatSnapshot(container.innerHTML);
      expect(html).toMatchSnapshot();
      unmount();
    });
  });
  afterEach(() => {
    localStorage.removeItem("environment");
  });
});
