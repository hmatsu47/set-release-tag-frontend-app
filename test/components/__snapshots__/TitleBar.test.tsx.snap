// Vitest Snapshot v1

exports[`<TitleBar /> > テスト環境Postfix 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic MuiAppBar-root MuiPaper-root css-xxxxxxxx\\"><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">テスト環境Postfix・コンテナリリース設定</div></div>"`;

exports[`<TitleBar /> > 本番環境Postfix 1`] = `"<div class=\\"MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic MuiAppBar-root MuiPaper-root css-xxxxxxxx\\"><div class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-root css-xxxxxxxx\\">本番環境Postfix・コンテナリリース設定</div></div>"`;
